Upstream: to be submitted...

From 8f871507713beee9ac8bd924aa7019ef140fff22 Mon Sep 17 00:00:00 2001
From: Kylie McClain <somasis@exherbo.org>
Date: Sun, 10 Apr 2016 18:15:35 -0400
Subject: [PATCH] Use va_copy instead of VA_COPY

VA_COPY is a glibc-specific function, va_copy is the same function, but
the name is defined by standard and should be available on all POSIX
libcs.
---
 src/bfu/msgbox.c    | 2 +-
 src/main/event.c    | 2 +-
 src/util/snprintf.c | 6 +++---
 src/util/snprintf.h | 6 +++---
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/bfu/msgbox.c b/src/bfu/msgbox.c
index d7af62b..aa12608 100644
--- a/src/bfu/msgbox.c
+++ b/src/bfu/msgbox.c
@@ -100,7 +100,7 @@ msg_text_do(unsigned char *format, va_list ap)
 	int infolen, len;
 	va_list ap2;
 
-	VA_COPY(ap2, ap);
+	va_copy(ap2, ap);
 
 	infolen = vsnprintf(NULL, 0, format, ap2);
 	info = mem_alloc(infolen + 1);
diff --git a/src/main/event.c b/src/main/event.c
index 9b3bc5d..0190e35 100644
--- a/src/main/event.c
+++ b/src/main/event.c
@@ -162,7 +162,7 @@ trigger_event_va(int id, va_list ap_init)
 		enum evhook_status ret;
 		va_list ap;
 
-		VA_COPY(ap, ap_init);
+		va_copy(ap, ap_init);
 		ret = ev_handler->callback(ap, ev_handler->data);
 		va_end(ap);
 
diff --git a/src/util/snprintf.c b/src/util/snprintf.c
index 7e9b286..d2a4bcb 100644
--- a/src/util/snprintf.c
+++ b/src/util/snprintf.c
@@ -178,7 +178,7 @@ dopr(char *buffer, size_t maxlen, const char *format, va_list args_in)
 	size_t currlen;
 	va_list args;
 
-	VA_COPY(args, args_in);
+	va_copy(args, args_in);
 
 	state = DP_S_DEFAULT;
 	currlen = 0;
@@ -825,7 +825,7 @@ elinks_vasprintf(char **ptr, const char *format, va_list ap)
 	int ret;
 	va_list ap2;
 
-	VA_COPY(ap2, ap);
+	va_copy(ap2, ap);
 
 	ret = vsnprintf(NULL, 0, format, ap2);
 	if (ret <= 0) return ret;
@@ -842,7 +842,7 @@ elinks_vasprintf(char **ptr, const char *format, va_list ap)
 	(*ptr) = (char *) malloc(ret + 1);
 	if (!*ptr) return -1;
 
-	VA_COPY(ap2, ap);
+	va_copy(ap2, ap);
 
 	return vsnprintf(*ptr, ret + 1, format, ap2);
 }
diff --git a/src/util/snprintf.h b/src/util/snprintf.h
index 1f3435a..a8655bd 100644
--- a/src/util/snprintf.h
+++ b/src/util/snprintf.h
@@ -11,11 +11,11 @@
 
 /* XXX: This is not quite the best place for it, perhaps. But do we have
  * a better one now? --pasky */
-#ifndef VA_COPY
+#ifndef va_copy
 #ifdef HAVE_VA_COPY
-#define VA_COPY(dest, src) __va_copy(dest, src)
+#define va_copy(dest, src) __va_copy(dest, src)
 #else
-#define VA_COPY(dest, src) (dest) = (src)
+#define va_copy(dest, src) (dest) = (src)
 #endif
 #endif
 
